nesta/core/scripts/docker_recipes/py37_amzn2_pytorch/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM amazonlinux:2
+
+# For debugging (leaving this here since we're still near the docker diskspace thresholds)
+RUN du -cxh --threshold=5M --max-depth=3 /
+
+RUN mkdir /root/.conda
+ENV PYTHONDONTWRITEBYTECODE=true
+ENV PYTHONIOENCODING=utf8
+ENV LANG=en_US.UTF-8
+
+# Install common dependencies
+RUN rm -rf /var/cache/yum/x86_64/latest
+RUN yum update -y
+RUN yum install aws-cli -y
+RUN yum install unzip -y
+RUN yum install which -y
+RUN yum install findutils -y
+RUN yum install wget -y
+RUN yum install gcc -y
+RUN yum clean all -y
+
+# Install base conda env
+RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh && /bin/bash ~/miniconda.sh -b -p /opt/conda && rm ~/miniconda.sh && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc && echo "conda activate base" >> ~/.bashrc && /bin/bash ~/.bashrc
+ENV PATH /opt/conda/bin:$PATH
+RUN conda config --add channels conda-forge
+RUN conda create -y -n py37 \
+    python=3.7 \
+    sentence-transformers \
```

Comment:
not really, I was always under the impression that the dependency management was done better but I never really tested the difference. It works as is: but do you see a reason to change it? (e.g. bloatiness?)

Reply:
Conda is slow and always seems to break but has always been something to put up with but now I struggle to see the point in it anymore...
Python wheels remove the need for conda in a lot of cases, and when there aren't wheels then we can get consistent system dependencies in Docker anyway.

Furthermore, `pip install`'ing on top of a conda environment undoes a lot of the conda dependency management anyway because `pip` will fetch the versions it wants for the packages it has been told to store.

Reply:
I say this as someone who always advocated for Conda because Docker has a higher barrier to entry, but the amount of time I lose in the average month due to conda randomly breaking / slow dependency resolution / slow rebuilding of environments etc. can be quite significant.

And lets not even go into

Reply:
ok, fair criticism. I've put this into `daps_utils` as an [issue](https://github.com/nestauk/daps_utils/issues/32) so that the default env doesn't have `conda`. Migrating from `nesta/nestauk` to `mono_daps` (or whatever we call it) will start in the next few months and so we can ditch at that point. Sound ok?

The docker setup in `daps_utils` is slimmer as well, as it uses a base image so that none of the duplication/boilerplate stuff is needed

Reply:
I think https://python-poetry.org/ is possibly worth a look into